packages/test-kit/src/sync-base-fs-contract.ts

```diff
@@ -409,6 +409,107 @@ export function syncBaseFsContract(testProvider: () => Promise<ITestInput<IBaseF
+
+                fs.renameSync(sourcePath, destinationPath)
+
+                expect(fs.statSync(destinationPath).isDirectory()).to.equal(true)
+                expect(fs.readFileSync(join(destinationPath, 'file'))).to.eql(SAMPLE_CONTENT)
+                expect(() => fs.statSync(sourcePath)).to.throw('ENOENT')
+            })
+
+            it('throws if source path doesn\'t exist', () => {
+                const { fs, tempDirectoryPath } = testInput
+                const { join } = fs.path
+                const sourcePath = join(tempDirectoryPath, 'file')
+
+                expect(() => fs.renameSync(sourcePath, join(tempDirectoryPath, 'file2'))).to.throw('ENOENT')
+            })
+
+            it('throws if the containing directory of the source path doesn\'t exist', () => {
+                const { fs, tempDirectoryPath } = testInput
+                const { join } = fs.path
+                const sourcePath = join(tempDirectoryPath, 'unicorn', 'file')
+
+                expect(() => fs.renameSync(sourcePath, join(tempDirectoryPath, 'file2'))).to.throw('ENOENT')
+            })
+
+            it('throws if destination containing path doesn\'t exist', () => {
+                const { fs, tempDirectoryPath } = testInput
+                const { join } = fs.path
+                const sourcePath = join(tempDirectoryPath, 'file')
+
+                fs.writeFileSync(sourcePath, SAMPLE_CONTENT)
+
+                expect(() => fs.renameSync(sourcePath, join(tempDirectoryPath, 'file2'))).not.to.throw('ENOENT')
+                expect(() => fs.renameSync(sourcePath, join(tempDirectoryPath, 'dir', 'file2'))).to.throw('ENOENT')
+            })
+
+            it('doesn\'t throw if destination path already exists for copying a directory over a non-existing directory', () => { //tslint:disable-line
+                const { fs, tempDirectoryPath } = testInput
+                const { join } = fs.path
+                const sourcePath = join(tempDirectoryPath, 'sourceDir')
+
+                fs.mkdirSync(sourcePath)
+                fs.writeFileSync(join(sourcePath, 'file'), SAMPLE_CONTENT)
+
+                expect(() => fs.renameSync(sourcePath, join(tempDirectoryPath, 'destDir'))).not.to.throw('EEXIST')
+            })
+
+            it('doesn\'t throw if destination path already exists for copying a directory over an empty directory', () => { //tslint:disable-line
+                const { fs, tempDirectoryPath } = testInput
+                const { join } = fs.path
+                const sourcePath = join(tempDirectoryPath, 'sourceDir')
+                const destPath = join(tempDirectoryPath, 'destDir')
+
+                fs.mkdirSync(sourcePath)
+                fs.mkdirSync(destPath)
+                fs.writeFileSync(join(sourcePath, 'file'), SAMPLE_CONTENT)
+
+                expect(() => fs.renameSync(sourcePath, destPath)).not.to.throw('EEXIST')
+            })
+
+            it('updates the parent directory of a renamed entry', () => {
```

Comment:
This test verifies the parent behavior only for a specific fixture (renaming of a directory). Might be useful to `readdirSync` in all other tests (non-error ones) and remove this test. I'm not sure though, WDYT?

Reply:
I guess this is already indirectly tested for files (by reading them), so this is only to make sure directories stay consistent as well because it can potentially work for files and still get a directory's shadow entries wrong